mindsdb/integrations/handlers/dremio_handler/dremio_handler.py

```diff
@@ -0,0 +1,236 @@
+        self.parser = parse_sql
+        self.dialect = 'dremio'
+
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+
+        self.base_url = f"http://{self.connection_data['host']}:{self.connection_data['port']}"
+
+        self.connection = None
+        self.is_connected = False
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self) -> dict:
+        """
+        Set up the connection required by the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        headers = {
+            'Content-Type': 'application/json',
+        }
+
+        data = '{' + f'"userName": "{self.connection_data["username"]}","password": "{self.connection_data["password"]}"' + '}'
+
+        response = requests.post(self.base_url + '/apiv2/login', headers=headers, data=data, verify=False)
+
+        return {
+            'Authorization': '_dremio' + response.json()['token'],
+            'Content-Type': 'application/json',
+        }
+
+    def disconnect(self):
+        """
+        Close any existing connections.
+        """
+
+        self.is_connected = False
+        return
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check connection to the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        response = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            self.connect()
+            response.success = True
+        except Exception as e:
+            log.logger.error(f'Error connecting to Dremio, {e}!')
+            response.error_message = str(e)
+        finally:
+            if response.success is True and need_to_close:
+                self.disconnect()
+            if response.success is False and self.is_connected is True:
+                self.is_connected = False
+
+        return response
+
+    def native_query(self, query: str) -> StatusResponse:
+        """
+        Receive raw query and act upon it somehow.
+        Args:
+            query (str): query in native format
+        Returns:
+            HandlerResponse
+        """
+
+        need_to_close = self.is_connected is False
+
+        auth_headers = self.connect()
+        data = '{' + f'"sql": "{query}"' + '}'
+
+        try:
+            sql_result = requests.post(self.base_url + '/api/v3/sql', headers=auth_headers, data=data)
+
+            job_id = sql_result.json()['id']
+
+            if sql_result.status_code == 200:
+                log.logger.info('Job creation successful. Job id is: ' + job_id)
+            else:
+                log.logger.info('Job creation failed.')
+
+            log.logger.info('Waiting for the job to complete...')
+
+            job_status = requests.request("GET", self.base_url + "/api/v3/job/" + job_id, headers=auth_headers).json()[
+                'jobState']
+
+            while job_status != 'COMPLETED':
```

Comment:
Why we need this?

Reply:
We need this because when we submit a job to the Dremio server, we need to wait until it is completed for us to extract the results. If we try to get the results while the job is still running, it will return an error.